app.go

```diff
@@ -88,6 +93,22 @@ func CurrentApp() App {
 	return (val).(appContainer).current
 }
 
+// AppMetadata captures the build metadata for an application.
+//
+// Since: 2.2
+type AppMetadata struct {
+	// ID is the unique ID of this application, used by many distribution platforms.
+	ID string
+	// Name is the human friendly name of this app.
+	Name string
+	// Version represents the version of this application, normally following semantic versioning.
+	Version string
+	// Build is the build number of this app, some times appended to the version number.
+	Build int
+	// Icon contains, if present, a resource of the icon that was bundled at build time.
+	Icon Resource
```

Comment:
Doesn't this mean that each application now bundles the application icon twice?

Also, are we really sure that we want to include the icon this way? Seems like a very inefficient way to store it as a string and then read from Base64 each time we start the application.

Reply:
Where is the other place the icon is stored? Developers may have packaged the icon before this was possible, but we can move the app to use this icon by default I would think, so not packaged manually.

I cannot find another way to package it - the format put in must be something that can be serialised to the commandline args (a restriction of this process).

Reply:
I would vote for leaving the icon out for now and tackling that at a later date. Almost everyone has the icon packaged outside already, but that's only part of the problem that I see. If I remove that packaged icon and only rely on this metadata, I won't get any icon when I just build using `go build`.

Reply:
We could do it at another time I suppose yes.
I think we need to do either this or start shipping with icons in some other way - it's required to solve issues like #2592 and #936.

Reply:
Can we not just use the icon that is passed to `app.SetIcon()` (and perhaps pass that into the metadata)? I think we need a better solution than passing it as a whole string with `fyne package`.

Reply:
Yes, users can bundle the icon manually and call SetIcon(). The metadata is static and will be the fallback so SetIcon will override it.
But why is the manual process favourable to an automated one?
An additional consideration - the MD5 is around half the number of bytes of the embedded resource, so not sure that it's a bad thing.

Reply:
It's not necessarily that a manual process is better than an automated one. I'm simply saying that I think we need a better solution than including the icon as a string (and thus getting no icon when running `go build`) and parsing it on each application start. I don't know what you mean with MD5, that's a hashing algorithm (that doesn't change length).

Is it possible to do the Base64 decoding before adding it in, instead of after? If we do so instead, we at least avoid the slow reading and avoid essentially storing the icon twice in memory. But even if that works, I still think we need a better solution given the `go build` issue mentioned above.

Reply:
Sorry, by MD5 I meant Base64. It is not possible to add the raw bytes because this is LDFLAGS injecting a raw string into the source.

Reply:
Just for clarity, where in your app do you use the app icon that is required without the app being packaged?

Reply:
> Just for clarity, where in your app do you use the app icon that is required without the app being packaged?

I do use the icon inside the about page to display the application icon there. However, the big problem for me is that it means that, without packaging, the binaries will show no icon at all (or more like the "missing icon" icon) in the list of open applications. That will be very apparent on any posix operating systems (maybe on Windows as well, I can't remember).

This just seems like a step in the wrong direction to me. I'd suggest avoiding the icon inclusion here for now and try to make the Windows notifications just use the `app.SetIcon` icon instead.

Reply:
The step back is partly what you mention, but for me, the big issue is what I described above about the icon not being present in the operating system application list when you run it (unless you bundle a second icon or build using `fyne package`). As the `SetIcon()` call is used to give the icon to the operating system (so that it can show a correct icon among the open applications), that functionality would go missing without using the workarounds mentioned above.

Reply:
> As the `SetIcon()` call is used to give the icon to the operating system

This is highly platform specific so it's tough to talk in generalisations - however on most OS the app icon is provided by extra files in the package phase. The app icon is a fallback which is used primarily for app windows that don't have a specific icon. This is now picking up the packaged metedata as well. So from now on Linux window managers that use the window icon instead of the app icon will get this icon without going through the package phase - or requiring users to bundle data manually.